adapta/storage/distributed_object_store/datastax_astra/astra_client.py

```diff
@@ -0,0 +1,183 @@
+"""
+ DataStax Astra client driver.
+"""
+
+#  Copyright (c) 2023. ECCO Sneaks & Data
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+import base64
+import os
+import shutil
+import tempfile
+import uuid
+from dataclasses import fields
+from typing import Optional, Dict, TypeVar, Callable, Iterable, Any, Type, List
+
+from _socket import IPPROTO_TCP, TCP_NODELAY, TIPC_CONN_TIMEOUT, TCP_USER_TIMEOUT
+
+import pandas
+from cassandra import ConsistencyLevel
+from cassandra.auth import PlainTextAuthProvider
+from cassandra.cluster import Cluster, Session, RetryPolicy, ExecutionProfile, EXEC_PROFILE_DEFAULT
+from cassandra.cqlengine import columns
+from cassandra.cqlengine.columns import Column
+from cassandra.cqlengine.connection import set_session
+from cassandra.cqlengine.models import Model
+from cassandra.cqlengine.named import NamedTable
+from cassandra.policies import ExponentialReconnectionPolicy
+from cassandra.query import dict_factory
+
+from adapta import __version__
+
+TModel = TypeVar("TModel")
+
+
+#  Copyright (c) 2023. ECCO Sneaks & Data
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+
+class AstraClient:
+    """
+    DataStax Astra (https://astra.datastax.com) credentials provider.
+
+    EXPERIMENTAL API NOTICE: THIS API IS NOT FINALIZED AND IS A SUBJECT TO CHANGE. USE WITH CAUTION UNTIL IT GRADUATES TO STABLE.
+    """
+
+    def __init__(
+        self,
+        client_name: str,
+        keyspace: str,
+        secure_connect_bundle_bytes: Optional[str] = None,
+        client_id: Optional[str] = None,
+        client_secret: Optional[str] = None,
+        reconnect_base_delay_ms=1000,
+        reconnect_max_delay_ms=5000,
+        socket_connection_timeout=5000,
+        socket_read_timeout=180000,
+    ):
+        self._secure_connect_bundle_bytes = secure_connect_bundle_bytes or os.getenv("PROTEUS__ASTRA_BUNDLE_BYTES")
```

Comment:
Maybe, it's better to create new environment variables with name starting from `ADAPTA`?

Reply:
not really, we've discussed earlier there will be a change similar to one we have in .net lib when this will be inferred from a global constant (as there is no app domain concept in python). And pre-3.0 we'll have both `PROTEUS__` and `ADAPTA__` allowed